chapter_3/2_spectral_clustering.tex

```diff
@@ -153,13 +154,242 @@ \section{Spectral Clustering}
 	
 	\begin{theorem}[Variational Characterization]
 		Let $M \in Sym(\mathbb R^{n\times n})$\\
-		$\lambda_1 \leq ... \leq \lambda_n$ eigenvalues of M\\
-		$\lambda_1 = min_{v \in S^{n-1}} Rank (v)$ let $v_1 = argmin$\\
-		$\vdots$\\
-		$\lambda_i = min_{\substack{ v \in S^{n-1}\\
-				v \perp < v_1 ... v_{i-1} >}} $
+		$\lambda_1 \leq ... \leq \lambda_n$ eigenvalues of $M$ \\
+		$\lambda_1 = \min_{v \in S^{n-1}} Rank (v)$\\
+		$\vdots$ \\
+		$\lambda_i = \min_{\substack{ v \in S^{n-1}\\
+				v \perp < v_1 ... v_{i-1} >}} Rank (v)$ \\
+		$\vdots$ \\
+		$\lambda_m = \max_{v \in S^{n-1}} Rank (v)$
 	\end{theorem}
-	Pf.
+	
+	\subsection{Spectral Clustering Algorithm}
+	
+	After introducing the graph Laplacian, we present the spectral clustering algorithm that projects data onto lower dimension space. The algorithm needs predefined number of clusters, and the similarity matrix.
+	
+	\textbf{Input:} $S$: $n\times n$ similarity matrix (on $n$ datapoints), $k$: number of clusters.
```

Comment:
Try to put this in a \begin{algorithm}\begin{algorithmic} environment if possible (see how its done elsewhere in the tex).